e2e/projects.spec.ts

```diff
@@ -0,0 +1,276 @@
+import { execSync } from 'child_process';
+import { test, expect } from '@playwright/test';
+import { SuperBlocks } from '../shared/config/superblocks';
+import tributePageHtml from './fixtures/tribute-page-html.json';
+import tributePageCss from './fixtures/tribute-page-css.json';
+import curriculum from './fixtures/js-ads-projects.json';
+import { authedPost } from './utils/request';
+
+import {
+  focusProjectEditor,
+  getProjectEditors,
+  clearProjectEditor
+} from './utils/project-editor';
+
+interface Meta {
+  challengeOrder: { id: string; title: string }[];
+}
+
+interface File {
+  contents: unknown;
+  fileKey: string;
+}
+
+type Solution = File[];
+
+interface Challenge {
+  title: string;
+  block: string;
+  superBlock: string;
+  dashedName: string;
+  solutions: Solution[];
+  isPrivate?: boolean;
+}
+
+interface block {
+  [key: string]: {
+    meta: Meta;
+    challenges: Challenge[];
+  };
+}
+
+const pythonProjects = {
+  superBlock: SuperBlocks.MachineLearningPy,
+  block: 'machine-learning-with-python-projects',
+  challenges: [
+    {
+      slug: 'book-recommendation-engine-using-knn',
+      nextChallengeText: 'Linear Regression Health Costs Calculator'
+    },
+    {
+      slug: 'cat-and-dog-image-classifier',
+      nextChallengeText: 'Book Recommendation Engine using KNN'
+    },
+    {
+      slug: 'linear-regression-health-costs-calculator',
+      nextChallengeText: 'Neural Network SMS Text Classifier'
+    },
+    {
+      slug: 'neural-network-sms-text-classifier',
+      nextChallengeText: 'Find the Symmetric Difference'
+    },
+    {
+      slug: 'rock-paper-scissors',
+      nextChallengeText: 'Cat and Dog Image Classifier'
+    }
+  ]
+};
+
+test.describe('Projects', () => {
+  test('Should be possible to submit Python projects', async ({ page }) => {
+    const { superBlock, block, challenges } = pythonProjects; // Ensure these are defined or imported
+
+    for (const { slug } of challenges) {
+      const url = `/learn/${superBlock}/${block}/${slug}`;
+      await page.goto(url);
+      await page
+        .locator('#dynamic-front-end-form #solution')
+        .fill('https://replit.com/@camperbot/python-project#main.py');
+
+      await page.locator("text=I've completed this challenge").click();
+      await expect(page.locator('text=go to next challenge')).toBeVisible();
+    }
+  });
+});
+
+test.describe('JavaScript projects can be submitted and then viewed in /settings and on the certifications', () => {
+  test.use({ storageState: 'playwright/.auth/development-user.json' });
+  test.beforeAll(() => {
+    execSync('node ./tools/scripts/seed/seed-demo-user');
+  });
+
```

Comment:
```suggestion
  test.skip(
    ({ browserName }) => browserName !== 'chromium',
    'Only chromium allows us to use the clipboard API.'
  );

```
So we can paste below.